src/components/gitOps/GitOpsConfiguration.tsx

```diff
@@ -190,267 +201,536 @@ class GitOpsConfiguration extends Component<GitOpsProps, GitOpsState> {
-            active: true,
+    suggestedValidGitOpsUrl() {
+        let gitOpsUrl = this.state.form.host
+        let suggestedValidGitOpsUrl: string
+        ShortGitHosts.forEach((shortGitHost) => {
+            if (gitOpsUrl.indexOf(shortGitHost) >= 0) {
+                suggestedValidGitOpsUrl = 'https://' + shortGitHost + '/'
+            }
+        })
+        return suggestedValidGitOpsUrl
+    }
+
+    isValidGitOpsUrl() {
+        if (!this.state.form.host) {
+            return true
         }
-        let promise = payload.id ? updateGitOpsConfiguration(payload) : saveGitOpsConfiguration(payload);
-        promise.then((response) => {
-            let resp = response.result
-            let errorMap = resp.stageErrorMap;
-            if (errorMap != null && Object.keys(errorMap).length == 0) {
-                this.props.handleChecklistUpdate('gitOps')
-                toast.success("Configuration validated and saved successfully");
-                this.setState({ validationStatus: VALIDATION_STATUS.SUCCESS, saveLoading: false, isFormEdited: false, deleteRepoError : resp.deleteRepoFailed});
-                this.fetchGitOpsConfigurationList();
-            } else {
-                this.setState({ validationStatus: VALIDATION_STATUS.FAILURE, saveLoading: false, isFormEdited: false, validationError : errorMap || [], deleteRepoError : resp.deleteRepoFailed});
-                toast.error("Configuration validation failed");
+
+        let isUrlAccepted: boolean = true
+        ShortGitHosts.forEach((shortGitHost) => {
+            if (this.state.form.host.indexOf(shortGitHost) >= 0) {
+                isUrlAccepted =
+                    (this.state.form.host.endsWith(shortGitHost + '/') ||
+                        this.state.form.host.endsWith(shortGitHost)) &&
+                    this.state.form.host.startsWith('https')
             }
-        }).catch((error) => {
-            showError(error);
-            this.setState({ view: ViewType.ERROR, statusCode: error.code, saveLoading: false });
         })
+
+        return isUrlAccepted
     }
 
-    validateGitOps(tab) {
+    saveGitOps() {
+        if (this.isInvalid()) {
+            toast.error('Some Required Fields are missing')
+            return
+        }
 
-        let isInvalid = this.isInvalid();
-        if (isInvalid) {
-            toast.error("Some Required Fields are missing");
-            return;
+        const isValidGitOpsUrl = this.isValidGitOpsUrl()
+        if (!isValidGitOpsUrl) {
+            this.setState({
+                isUrlValidationError: true,
+                validationStatus: VALIDATION_STATUS.DRY_RUN,
+                saveLoading: false,
+                validateLoading: false,
+            })
+        } else {
+            this.setState({
+                isUrlValidationError: false,
+                validationStatus: VALIDATION_STATUS.LOADER,
+                saveLoading: true,
+            })
+            const payload = {
```

Comment:
is this defined as type? if yes, specify type here

Reply:
form to payload setting should be extracted out. we are doing it at multiple places

Reply:
done